### ISSUE TITLE
awaitRace does not really race()

### ISSUE THREAD

**Issue by clue** [clue/php-block-react#6]
The currently implementation of `awaitRace($promises)` essentially works like this:
```
awaitOne(Promise\any($promises))
```

The naming *implies* this might call the equivalent of this:

```
awaitOne(Promise\race($promises))
```

IMO both have valid use cases and might be worth implementing.

See also #5 for some background on the functional API.

At the very least, this needs some documentation. A better method name would also be appreciated.

**Comment by clue** [comment]
> A better method name would also be appreciated.

PR #5 simply renames this method to reflect what it actually does: `awaitAny()`

**Comment by clue** [comment]
FWIW: We should eventually also provide a *real* `awaitRace()` function. This will likely be postponed until after #5 is ready.
